client/verta/tests/unit_tests/registry/test_check_model_dependencies.py

```diff
@@ -49,28 +41,28 @@ def test_check_model_dependencies_complete(dependency_testing_model, complete_en
 
 
 def test_check_model_dependencies_missing_raise(dependency_testing_model, complete_env) -> None:
-    """ Verify that check_model_dependencies raises an exception for missing
-    packages when `raise_for_missing` is True.
+    """ Verify that check_model_dependencies raises an exception, with the
+    correct message, for missing packages when `raise_for_missing` is True.
     """
     incomplete_env = Python(
-        [ r for r in complete_env.requirements if r != 'click==0.0.1' ]
+        [r for r in complete_env.requirements if r != 'click==0.0.1']
     )  # drop a single dependency to be caught
     with pytest.raises(RuntimeError) as err:
         _check_model_dependencies(
             model_cls=dependency_testing_model,
             environment=incomplete_env,
             raise_for_missing=True,
         )
-    assert err.value.args[0] == "the following packages are required by the model " \
-                                "but missing from the environment: {'click'}"
+    assert err.value.args[0] == "the following packages are required by the model but missing " \
+                                "from the environment:\nclick (installed via ['click'])"
 
 
 def test_check_model_dependencies_missing_warning(dependency_testing_model, complete_env) -> None:
-    """ Verify that check_model_dependencies defaults to raising a warning for
-    missing packages when `raise_for_missing` is False.
+    """ Verify that check_model_dependencies defaults to raising a warning, with
+    the correct message, for missing packages when `raise_for_missing` is False.
     """
     incomplete_env = Python(
-        [ r for r in complete_env.requirements if r != 'pandas==0.0.1' ]
+        [r for r in complete_env.requirements if r not in ['pandas==0.0.1']]
```

Comment:
Having 2 or more missing packages in this test would have required more refactoring to deal with the ordering of the dictionary returned so that the error message can be tested.  (Ordering of the missing packages in the error message is random because each is pulled from a dict).  We already have explicit coverage of "module name != pkg name" in `test_model_dependencies.test_package_names()`, but there's no harm in swapping in `yaml` for `pandas` in this test.

Reply:
Just added that change.  Even though the util function is tested, it's nice to see the final step tested here as well.

Reply:
It can be a good practice to sort items before inserting them in the error message, to make things more deterministic.

We should want to cover cases with multiple missing packages.

Reply:
And as an aside,

> Ordering of the missing packages in the error message is random because each is pulled from a dict

It's not quite accurate to say it's random; Python dictionaries maintain order.